src/main/java/org/opensearch/security/dlic/rest/validation/RequestContentValidator.java

```diff
@@ -286,14 +286,6 @@ public XContentBuilder toXContent(XContentBuilder builder, ToXContent.Params par
                         .get(SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE, "Password does not match minimum criteria")
                 );
                 break;
-            case WEAK_PASSWORD:
-            case SIMILAR_PASSWORD:
-                builder.field("status", "error");
-                builder.field(
-                    "reason",
-                    validationContext.settings().get(SECURITY_RESTAPI_PASSWORD_VALIDATION_ERROR_MESSAGE, validationError.message())
```

Comment:
Why remove this message, it was useful?

Reply:
@peternied I removed this message because it can be misleading in some situations. For example:
- Set validation message = 'Password must be longer than 5 char'
- Try to create user with password 'aaaaaa'
- API returns 'Password must be longer than 5 char', even though this is already true, and it is actually just weak

By removing this case statement, weak and similar passwords to the username will instead be caught in the default statement, which would return a more specific message in the API, such as "Weak password", or "Similar password"

Reply:
OK, I've looked into this - the way validation errors are used with the customizable messages is convoluted.  Lets not try to fix that right now.  Since no existing tests were broken lets move forward and we can come back to this later.

Thanks for looking into this issue

Root problem is that `ValidationError` is an enum with hardcoded messages these custom messages should have an alternative channel (e.g. CustomizableValidationMessage) to be populated at the source rather than that strange toXContent function,

Reply:
> Since no existing tests were broken

And I just checked the CI to see that this wasn't the case, look at https://github.com/opensearch-project/security/actions/runs/5810367794

@derek-ho can you look into these failures and propose how you'd like to move forward?